src/pal/src/exception/signal.cpp

```diff
@@ -554,6 +555,23 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
         ASSERT("sigprocmask failed; error is %d (%s)\n",errno, strerror(errno));
     } 
 
+#if defined(_AMD64_)
+    if (pointers->ExceptionRecord->ExceptionCode == EXCEPTION_INT_DIVIDE_BY_ZERO)
+    {
+        // It is possible that an overflow may also have been mapped to a divide-by-zero
+        // exception. This happens when we try to divide the maximum negative value of a
+        // signed integer with -1. 
+        //
+        // Thus, we will attempt to decode the instruction @ RIP to determine if that
+        // is the case using the faulting context.
+        if (IsDivByZeroAnIntegerOverflow(pointers->ContextRecord, pointers->ExceptionRecord))
```

Comment:
It may be better to do this adjustment only after we have determined that the exception happened in managed code (e.g. in MapWin32FaultToCOMPlusException). Disassembling somebody else's code is potentially dangerous.